### DISCUSSION TITLE
Adding Solokeys2 in macOS 13.2 fails on the last screen

### DISCUSSION THREAD

**Discussion by tehabe** [solokeys/solo2#154] [General]
Today Apple released macOS 13.2 with the ability to add security keys to your Apple ID. I tried this with my two Solo keys and it fails at the last screen. It feels like Safari with is also impossible to add security keys to almost any website. It is very confusing what it not working here. Maybe Apple checks what kind of device it is and doesn't know it.

Anyway, I fails on this screen. I click "Done" and something rotates for a few seconds and nothing happens, no error message, and no way out, I only can force quit the System Settings app.

This is really the only thing which annoys me of the Solo keys at the moment and I don't know if the bug is in macOS or Solo's firmware.

<img width="478" alt="Screenshot 2023-01-23 at 22 38 18" src="https://user-images.githubusercontent.com/67130363/214163700-b0d448c1-086f-4990-aa3c-f9f55906bb10.png">

**Comment by urbaniak** [comment]
Found the same on macOS using one yubikey and solo2, iOS and iPadOS is shows "unexpected error" like on the screenshot below.

![IMG](https://user-images.githubusercontent.com/25346/214167774-439ef919-4fd8-42f3-8743-3f1f338b3414.png)

**Comment by tehabe** [comment]
So if even the great Yubikeys do not work…

**Comment by teslamint** [comment]
Solo v1 works after adding first 2 security key with 2 Yubikeys, but not any Solo v2 works by now.

**Comment by urbaniak** [comment]
It worked for me with solo1 + yubico, when trying to register solo2 getting `HTTPError: 500`, so adding solo2 probably crashes apple's backend.

<img width="472" alt="Screenshot 2023-01-24 at 9 06 12 am" src="https://user-images.githubusercontent.com/25346/214240974-b5a5e1ae-5887-4e29-a810-37a337e9aea2.png">

**Reply by msiqx9** [reply]
I added 6 keys (Yubico, Feitian, KEY-ID and Solo v1 Tap) via my iPad 6th gen with iPadOS 16.3.  All keys including Solo v1 Tap were accepted.  My Solo v2 was rejected by Apple.  I think this is an Apple policy decision (Solo v2 key not on Apple's approved key list) rather than any defect with the Solo v2 key.

**01-04-2023 update** - I succeeded in adding my Solo v2 key to my Apple ID via my iPad 10th gen with iPadOS 16.4.

**Reply by alexjmoore** [reply]
I wouldn't be surprised if it is a solo2 issue though, it barely worked on windows initially (although after a long time they did eventually patch that) and it has never worked on Android to date (at least not for me) again a thread on that with no response.

**Comment by ab-gh** [comment]
Has anyone had further success with using Solo2 for their Apple ID? I also get to the last step before an "unexpected error", on iOS 17.1.1, Sonoma 14.1.1, with firmware 1:20200101.9

**Reply by msiqx9** [reply]
I just removed and re-added my Solo v2 key to my Apple ID using a MacBook on Sonoma 14.1.2.  My Solo v2 key has the current firmware 2:20220822.0
